test/e2e/autoscale_test.go

```diff
@@ -314,19 +328,24 @@ func assertAutoscaleUpToNumPods(ctx *testContext, numPods int32) {
 
 	assertNumberOfPodsEvery(2*time.Second, ctx, errChan, stopChan, minPods, maxPods)
 
-	if err := generateTraffic(ctx, int(numPods*10), 30*time.Second); err != nil {
-		ctx.t.Fatalf("Error during scale up: %v", err)
-	}
+	go generateTraffic(ctx, int(numPods*10), timeout, errChan, stopChan)
```

Comment:
I see, you refactored `generateTraffic` to pass the errChan here. Personally, I'd prefer to not pass this channel around too much but rather keep `generateTraffic` the way it is (regarding return error). That'd change this piece of code slightly to:

```golang
go func() {
	if err := generateTraffic(ctx, int(numPods*10), timeout, stopChan)); err != nil {
		errChan <- err
	}
}
```

But simplifies `generateTrafficSync` and generally is a bit more straightforward with the channel usage. WDYT?

Reply:
thanks for the suggestion, it does seem cleaner with your approach.